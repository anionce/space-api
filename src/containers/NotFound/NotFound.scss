.not-found-container {
	display: flex;
	justify-content: center;
	flex-direction: column;
	align-items: center;

	img {
		margin-top: 20px;
		border-radius: 10px;
		max-width: 200px;
		height: auto;
	}

	.text {
		font-size: 40px;
		font-weight: 600;
		font-family: Roboto;
	}
}
