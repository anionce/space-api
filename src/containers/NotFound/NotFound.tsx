import React from 'react';
import { useTranslation } from 'react-i18next';
import notFound from '../../assets/img/notFound.svg';
import './NotFound.scss';

export const NotFound = () => {
	const { t } = useTranslation();

	return (
		<div className='not-found-container'>
			<h1 className='text'>{t('space_api.not-found.title')}</h1>
			<img alt='not-found' src={notFound}></img>
		</div>
	);
};
