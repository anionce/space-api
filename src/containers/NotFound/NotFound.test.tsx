import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { NotFound } from './NotFound';

test('renders correcly', () => {
	const { container } = customRender(<NotFound />);

	expect(container).toMatchSnapshot();
});

test('all elements are displayed', () => {
	customRender(<NotFound />);

	expect(screen.getByAltText('not-found')).toBeInTheDocument();
	expect(screen.getByText(/Sorry, this doesn't exist/i)).toBeInTheDocument();
});
