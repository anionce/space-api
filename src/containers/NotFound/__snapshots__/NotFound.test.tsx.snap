// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="not-found-container"
  >
    <h1
      class="text"
    >
      Sorry, this doesn't exist
    </h1>
    <img
      alt="not-found"
      src="notFound.svg"
    />
  </div>
</div>
`;
